### PR TITLE
USAGOV-1482: recaptcha v3 upgrade

### PR DESCRIPTION
**Pull request by developer43**
## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-1482

## Description

Replace recaptcha v2 with recaptcha v3
## Type of Changes

- [ ] New Feature
- [ ] Bugfix
- [x] Frontend (Twig, Sass, JS)
  - Add screenshot showing what it should look like
- [ ] Drupal Config (requires "drush cim")
- [ ] New Modules (requires rebuild)
- [ ] Documentation
- [ ] Infrastructure
  - [ ] CMS
  - [ ] WAF
  - [ ] Egress
  - [ ] Tools
- [ ] Other

## Testing Instructions

To test goto http://localhost/site-issue-report-form and verify the recaptcha shows in the bottom right corner of the page.
Screenshot 2024-08-05 at 11 40 39 AM

then submit the form without filling out the fields to verify the error validation works.
Screenshot 2024-08-05 at 1 55 14 PM

open the dev tools for the page and goto the "Network" tab and fill out the form and submit. Select the webservlet in the 
Screenshot 2024-08-02 at 1 40 48 PM

Selet the "Form Data" tab and verify the data you submitted is there along with the g-recaptcha-response data
Screenshot 2024-08-02 at 1 40 34 PM

Repeat these steps for the Spanish page http://localhost/es/reporte-problemas-en-este-sitio-web

### Change Requirements

- [ ] Requires New Documentation (Link: {})
- [ ] Requires New Config
- [ ] Requires New Content

### Validation Steps

- Test instruction 1
- Test instruction 2
- Test instruction 3

## Security Review

- [ ] Adds/updates software (including a library or Drupal module)
- [ ] Communication with external service
- [ ] Changes permissions or workflow
- [ ] Requires SSPP updates

## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.

### CODE REVIEW

**Comment by akf** [comment]
@developer43 sorry I didn't review this right away. Now there's a merge conflict with dev to resolve. It's pretty obvious ("defer" has been added to a couple of script calls on dev, I think), but it's better if you make the update to the PR.

File: web/themes/custom/usagov/scripts/reportAProblem.js [hunk 1]
```diff
@@ -276,64 +157,26 @@ var submitPressed = function () {
   "use strict";
 
   // reCaptcha Validation and error styling.
-  var reCaptchaValidationResult = reCaptchaValidation();
+  // var reCaptchaValidationResult = reCaptchaValidation();
```
Comment (akf) [inline web/themes/custom/usagov/scripts/reportAProblem.js]: Please remove this line (and the comment above it), now that you know we don't need them. And also the commented-out line just before the return statement.

Comment (akf) [inline web/themes/custom/usagov/scripts/reportAProblem.js]: Please remove this line (and the comment above it), now that you know we don't need them. And also the commented-out line just before the return statement.

File: web/themes/custom/usagov/templates/field--node--field-custom-twig-content-reporte-problemas-en-este-sitio-web.html.twig [hunk 2]
```diff
@@ -25,16 +25,16 @@
...
 			id="myform"
 			method="post"
 			name="myform"
 			novalidate>
 			<div class="padding-1px bg-gray-5">
 				<div class="maxw-mobile margin-x-auto margin-y-4">
-					<input type="hidden" name="captcha_settings" value='{"keyname":"Recaptcha_FCIC","fallback":"true","orgId":"00DU0000000Leux","ts":""}'/>
-					<input type="hidden" name="orgid" value="00DU0000000Leux"/>
+					<input type=hidden name='captcha_settings' value='{"keyname":"RecaptchaV3","fallback":"true","orgId":"00DU0000000Leux","ts":""}'>
+					<input type=hidden name="orgid" value="00DU0000000Leux">
```
Comment (akf) [inline web/themes/custom/usagov/templates/field--node--field-custom-twig-content-reporte-problemas-en-este-sitio-web.html.twig]: I wouldn't have noticed this if it weren't highlighted as a difference, but there should be quotation marks around "hidden".

File: web/themes/custom/usagov/templates/field--node--field-custom-twig-content-reporte-problemas-en-este-sitio-web.html.twig [hunk 3]
```diff
@@ -144,5 +120,5 @@
 <p class="clearfix"></p>
 
 <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.4/jquery.min.js"></script>
-<script src="https://www.google.com/recaptcha/api.js?hl=es"></script>
+<script src="https://www.google.com/recaptcha/api.js"></script>
```
Comment (akf) [inline web/themes/custom/usagov/templates/field--node--field-custom-twig-content-reporte-problemas-en-este-sitio-web.html.twig]: Maybe we still need the ?hl=es query string. If you didn't already check, please take a look at whether the recaptcha badge has text in English on Spanish pages, and if so, if putting the query string back on changes it to Spanish.

**Review by akf — COMMENTED** [review]
Just commenting for now until you get a chance to resolve that merge conflict. I read through the code changes and made a few notes. I haven't actually loaded this locally and tested it out.

File: web/themes/custom/usagov/scripts/reportAProblem.js (hunk 1 above)
Comment (developer43) [reply web/themes/custom/usagov/scripts/reportAProblem.js]: Thanks @akf, I fixed the conflict.

File: web/themes/custom/usagov/templates/field--node--field-custom-twig-content-reporte-problemas-en-este-sitio-web.html.twig (hunk 3 above)
Comment (akf) [reply web/themes/custom/usagov/templates/field--node--field-custom-twig-content-reporte-problemas-en-este-sitio-web.html.twig]: Following up, it looks like we do want the query string. It causes the text on the recaptcha to come through in Spanish.

**Review by akf — APPROVED** [review]
I'm not sure I can approve this, since I pushed some changes myself, but here goes.

**Pull request merged**